tests/test_fastavro.py

```diff
@@ -2105,3 +2106,65 @@ def test_record_named_type():
     records = [{"test1": {"field1": "foo"}, "test2": {"field1": "bar"}}]
     parsed_schema = fastavro.parse_schema(schema)
     assert records == roundtrip(parsed_schema, records)
+
+
+def test_write_required_field_name():
+    '''
+    Test that when we can use tuple style of writing unions
```

Comment:
I don't think this comment matches the test since this test doesn't use tuple style writing